src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -49,307 +58,560 @@ public class GitLabPluginTest extends AbstractJUnitTest {
-        assertThat(branchIndexingLog, containsString("Scheduled build for branch: firstbranch"));
-        assertThat(branchIndexingLog, containsString("Scheduled build for branch: failedjob"));
-        assertThat(branchIndexingLog, containsString("Scheduled build for branch: MR-1"));
-        assertThat(branchIndexingLog, containsString("1 merge requests were processed"));
+    private void assertExistAndResult(final WorkflowJob job, final boolean withSuccess) {
+        final Build.Result expectedResult = withSuccess ? Build.Result.SUCCESS : Build.Result.FAILURE;
+        assertJobExists(job);
+        waitFor()
+                .withMessage("Waiting for job '%s' to complete with result %s", job, expectedResult)
+                .withTimeout(Duration.ofSeconds(time.seconds(BUILD_COMPLETION_TIMEOUT_SECONDS)))
+                .pollingEvery(Duration.ofSeconds(POLLING_INTERVAL_SECONDS))
+                .until(() -> {
+                    try {
+                        Build lastBuild = job.getLastBuild();
+                        if (lastBuild == null) {
+                            return false;
+                        }
+                        String result = lastBuild.getResult();
+                        return result != null && result.contains(expectedResult.name());
+                    } catch (Exception e) {
+                        return false;
+                    }
+                });
     }
 
-    private void configureJobWithGitLabBranchSource(
-            final WorkflowMultiBranchJob job, String adminUserName, String project) {
-        final GitLabBranchSource ghBranchSource = job.addBranchSource(GitLabBranchSource.class);
-        ghBranchSource.setOwner(adminUserName);
-        ghBranchSource.setProject(adminUserName, project);
+    private void assertJobExists(final WorkflowJob job) {
+        waitFor()
+                .withMessage("Waiting for job '%s' to be created", job)
+                .withTimeout(Duration.ofSeconds(time.seconds(JOB_CREATION_TIMEOUT_SECONDS)))
+                .pollingEvery(Duration.ofSeconds(POLLING_INTERVAL_SECONDS))
+                .until(() -> {
+                    try {
+                        job.url("").openStream().close();
+                        return true;
+                    } catch (IOException e) {
+                        return false;
+                    }
+                });
     }
 
-    private void assertExistAndResult(final WorkflowJob job, final boolean withSuccess) {
-        final Build.Result expectedResult = (withSuccess) ? Build.Result.SUCCESS : Build.Result.FAILURE;
-        waitFor(job, Matchers.pageObjectExists(), (int) time.seconds(3));
-        waitFor(job.getLastBuild().getResult(), Matchers.containsString(expectedResult.name()), (int) time.seconds(3));
+    // ==================== Git operations ====================
+
+    private void setupInitialBranchViaGit(String repoUrl, String branchName, String content) throws IOException {
```

Comment:
to be pedantic,
```suggestion
    private void setUpInitialBranchViaGit(String repoUrl, String branchName, String content) throws IOException {
```

since _setup_ is a noun whereas _set up_ is the (separable) verb